frontend_lib/src/helper.js

```diff
@@ -927,6 +928,213 @@ export const handleClickCopyLink = (contentId) => {
+    newHtml = newHtml.replace(mentionTag, mention)
+  })
+
+  return newHtml
+}
+
+const replaceHTMLElementWithMentionRole = (rolelist, html) => {
+  const mentionRegex = /<html-mention roleid="(\d+)"><\/html-mention>/g
+  const mentionTagList = html.match(mentionRegex)
+  if (!mentionTagList) return html
+
+  let newHtml = html
+
+  mentionTagList.forEach(mentionTag => {
+    const mentionTagData = mentionTag.match(/roleid="(\d+)"/)
+    const roleId = mentionTagData[1]
+
+    const role = rolelist.find(r => r.role_id === roleId)
+    const mention = `@${role.slug}`
+    newHtml = newHtml.replace(mentionTag, mention)
+  })
+
+  return newHtml
+}
+
+const replaceHTMLElementWithMentionUser = (userList, html) => {
+  const mentionRegex = /<html-mention userid="(\d+)"><\/html-mention>/g
+  const mentionTagList = html.match(mentionRegex)
+  if (!mentionTagList) return html
+
+  let newHtml = html
+
+  console.log('userList', userList)
+
+  mentionTagList.forEach(mentionTag => {
+    console.log('mentionTag', mentionTag)
+    const mentionTagData = mentionTag.match(/userid="(\d+)"/)
+    console.log('mentionTagData', mentionTagData)
+    const userId = mentionTagData[1]
+    console.log('userId', userId)
+
+    const user = userList.find(u => u.id === Number(userId))
+    let mention = ''
+    if (!user) {
+      console.warn(
+        `helper.js - replaceHTMLElementWithMentionUser - User from id ${userId} not found`
+      )
+      mention = `@UnknownUser`
+    } else {
+      mention = `@${user.username}`
+    }
+    newHtml = newHtml.replace(mentionTag, mention)
+  })
+
+  return newHtml
+}
+
+/**
+ * Replace html mention element with mention
+ * @param {List[role]} roleList List of role that can be mentionned
+ * @param {List[user]} userList List of user that can be mentionned
+ * @param {String} html Current content of the editor
+ * Example:
+ * - Input: `<p>Test <html-mention userid="151"/><html-mention></p>`
+ * - Output: `<p>Test @Jhon</p>`
+ */
+export const replaceHTMLElementWithMention = (rolelist, userList, html) => {
+  let newHtml = replaceHTMLElementWithMentionRole(rolelist, html)
+  newHtml = replaceHTMLElementWithMentionUser(userList, newHtml)
+  return newHtml
+}
+
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// NOTE - MP - 2022-12-09 - LINK SECTION
+// /////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Search content in a string
+ * @param {String} text The text to search mentions in
+ * @returns {List[String]} List of content id found
+ * Example:
+ * - Input: `<p>Test #844</p>`
+ * - Output: `['#844']`
+ */
+export const searchContent = (text) => {
+  // Regex explanation: https://regex101.com/r/z1WUUu/1
+  // Match (#XXX part): '#XXX', '#XXX ', ' #XXX', '#XXX:', ':#XXX', '(#XXX)', '#XXX!', ...
+  // Don't match: 'XXX#XXX', '#<span>XXX</span>'
+  const contentRegex = /(?<=^|\s|\W)#([0-9]+)\b/g
```

Comment:
This regex is currently wrong. It will match `"#XXX"` and cause the text sent after an edit where a content were already typed to be broken?

1. Create a comment with a content link
2. Edit the previous comment so it can be send again
3. See the broken link/text

Reply:
Changed to use `/(?<=^|\s|\W)(?<!title=")#([0-9]+)\b/g`
See https://regex101.com/r/z1WUUu/3